.github/workflows/ci.yml

```
@@ -74,6 +74,72 @@ jobs:
+        run: git checkout origin/main
+
+      - name: Run benchmarks on main branch
+        run: |
+          cd nexum_core
+          cargo bench --bench storage_bench -- --output-format json | tee main_bench.json
+          cargo bench --bench sql_bench -- --output-format json | tee -a main_bench.json
+          cargo bench --bench executor_bench -- --output-format json | tee -a main_bench.json
+          cargo bench --bench filter_bench -- --output-format json | tee -a main_bench.json
+
+      - name: Compare benchmark results
+        run: |
```

actionlint (1.7.9)

shellcheck reported issue in this script: SC2086:info:10:12: Double quote to prevent globbing and word splitting

(shellcheck)